internal/impl/devbox.go

```diff
@@ -827,6 +827,15 @@ func (d *Devbox) computeNixEnv(ctx context.Context, usePrintDevEnvCache bool) (m
 	nixEnvPath := env["PATH"]
 	debug.Log("PATH after plugins and config is: %s", nixEnvPath)
 
+	// We filter out nix store paths so that if a user removes a package from their devbox
+	// it no longer is available in their environment. This is needed because nix may keep
+	// packages around if they are used somewhere else or if the user hasn't triggered
+	// garbage collection.
+	nixEnvPath = filterPathList(nixEnvPath, func(path string) bool {
+		return !strings.HasPrefix(path, "/nix/store")
```

Comment:
A problem with filtering _all_ the `/nix/store` paths is that we also remove many utilities that are automatically included by `pkgs.mkShell` in the generated `flake.nix`.

As a result, basic functionality breaks, like invoking `mkdir`.

@mikeland86 thoughts on fixing this?

A (bad) idea is to generate `print-dev-env` PATH without any packages in the flake.nix. Keep the `/nix/store/` paths from this, while removing any new /nix/store paths added from print-dev-env using the generated flake.nix. BUT this would involve calling `print-dev-env` twice, which is slow.

Reply:
Can you filter out `$BuildInputs` only?

Reply:
yep, this works I think. PTAL!